### ISSUE TITLE
Build blog-pages in two languages

### ISSUE THREAD

**Issue by Anjade** [d4l-data4life/kirby3-static-site-generator#72]
This is the code i use for building static blog-pages in two languages.

`'routes' =>
  [
    [
        'pattern' => 'kirby/kirby/(:any)/notes/page:(:num)',
        'action' => function ($dummy, $value)
        {
            
            $pageId = "notes";
            $param = "page";
            
            $params = kirby()->request()->url()->params();
            $params->$param = $value;
            $html = page($pageId)->render();
            
  
            $params->$param = null; // reset the param after rendering the page
            return $html;
        },
    ],
  ],
  
  
  'd4l' => [
    'static_site_generator' => [
      'endpoint' => 'generate-static-site',
      'base_url' => '/kirby-static',
      'output_folder' => './kirby-static',
    ],
  ],

  'ready' => function ($kirby)
  {

      $notePerPage = 3; // change here and in /site/controllers/notes.php!
      $pagesCount = page('notes')->children()->listed()->count();
           
      $pageNumbers = intval($pagesCount/$notePerPage);
      
      if (($pagesCount % $notePerPage) != 0)
      {
        $pageNumbers++;
      }      

      for ($i = 2; $i <= $pageNumbers; $i++)
      {

        $myRoutes[] = [
          'path' => "en/notes/$i",
          'route' => "kirby/kirby/en/notes/page:$i",
          'languageCode' => 'de',
        ];

        $myRoutes[] = [
          'path' => "de/notes/$i",
          'route' => "kirby/kirby/de/notes/page:$i",
          'languageCode' => 'en',
        ];

      }

    return [
        'd4l.static_site_generator.custom_routes' => $myRoutes
    ];
  },`

This code made a lot of problems with the articles in my blog. The only way to have articles in blog-pages with the correct language is to change the languageCode in myRoutes[] to the wrong language (see above!). I have sent urls to kirby and static pages and FTP per mail.

**Comment by jonathan-reisdorf** [comment]
There are a lot of unconventional choices in the code posted (like the `kirby/kirby` prefix, the unused language parameter in the route, the way the pages count is calculated), but I can not see the error. You can remove the extra language from your route, it has no effect on the language set (that's determined with `languageCode`).

Btw. for generating the pagination range, I recommend something like
```php
array_slice(
  page("notes")
    ->children()
    ->listed()
    ->paginate($notePerPage)
    ->pagination()
    ->range(PHP_INT_MAX),
  1
)
```
That would give you an array with `2`, `3`, `4`, ... that you can then use in conjunction with `array_map` to generate your custom (static file) routes.

**Comment by jonathan-reisdorf** [comment]
I was now able to reproduce the behavior when generating pagination URLs. It seems that kirby uses a special url getter here (`$kirby->url("current")`) when using the pagination object, which would not work as expected in the static site generation context. I will investigate if this can be overridden

**Comment by jonathan-reisdorf** [comment]
Once the linked PR is merged, update to version `1.9.0` of the plugin. Something like this should then work:

In the routes:
```php
    [
      "pattern" => "generate-static-site/(:any)/page:(:num)",
      "action" => function ($pageId, $pageNum) {
        $params = kirby()
          ->request()
          ->url()
          ->params();
        $params->page = $pageNum;
        $html = page($pageId)->render();

        $params->page = null; // reset the param after rendering the page
        return $html;
      },
    ]
```

and in your example you'd switch the custom routes so they are like this:

```php
$myRoutes[] = [
  "path" => "de/notes/$i",
  "page" => "notes",
  "route" => "generate-static-site/notes/page:$i",
  "languageCode" => "de",
];
$myRoutes[] = [
  "path" => "en/notes/$i",
  "page" => "notes",
  "route" => "generate-static-site/notes/page:$i",
  "languageCode" => "en",
];
```

The new thing here is that you can now add a page to the custom route so that when rendering the route there is an additional page context which is used when rendering the pagination links.
You will still have to manually reformat the pagination links from `page:x` to `/x` in your template, but the base URL should now work and there should be no mixup of languages.

You can also completely skip adding a "proxy" route and instead just render the custom routes without pointing to a route. You can then add the page as a template variable by using the `data` attribute in the custom routes.

Example:
```php
$myRoutes[] = [
  "path" => "de/notes/$i",
  "page" => "notes",
  "languageCode" => "de",
  "data" => ["pageNum" => $i]
];
$myRoutes[] = [
  "path" => "en/notes/$i",
  "page" => "notes",
  "languageCode" => "en",
  "data" => ["pageNum" => $i]
];
```

Then, in your template you can get the current page number like this:
```
$pageNumber = $pageNum ?? param('page') ?? 1;
```

Then, when calling `->children()->paginate(` just pass it as the second argument to set the current page number. Documentation: https://getkirby.com/docs/reference/objects/cms/pages/paginate
